app/hotels/src/allHotels/SearchForm.js

```diff
@@ -0,0 +1,134 @@
+// @flow
+
+import * as React from 'react';
+import {
+  TextInput,
+  DatePicker,
+  Color,
+  Button,
+} from '@kiwicom/react-native-app-common';
+import { View, StyleSheet } from 'react-native';
+import type { Search } from './types';
+
+const styles = StyleSheet.create({
+  form: {
+    padding: 10,
+    backgroundColor: Color.brand,
+  },
+  destination: { marginBottom: 6 },
+  row: {
+    flexDirection: 'row',
+    height: 40,
+  },
+  datePicker: {
+    flex: 1,
+    marginRight: 10,
+  },
+});
+
+const buttonStyles = StyleSheet.create({
+  button: {
+    width: 115,
+    backgroundColor: '#eee',
+    flex: 1,
+    borderRadius: 0,
+  },
+  buttonText: {
+    color: Color.grey._900,
+  },
+});
+
+type Props = {
+  onChange: (search: Search) => void,
+};
+type State = {
+  destination: string | null,
+  latitude: number | null,
+  longitude: number | null,
+  checkin: Date | null,
+  checkout: Date | null,
+  roomsConfiguration: {
+    adultsCount: number,
+  },
+};
+
+export default class SearchForm extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      destination: null,
+      latitude: null,
+      longitude: null,
+      checkin: null,
+      checkout: null,
+      roomsConfiguration: {
+        adultsCount: 1,
+      },
+    };
+  }
+
+  _handleOnChange = () =>
+    this.props.onChange({
+      latitude: this.state.latitude,
+      longitude: this.state.longitude,
+      checkin: this.state.checkin,
+      checkout: this.state.checkout,
+      roomsConfiguration: this.state.roomsConfiguration,
+    });
+
+  _handleDestinationChange = destination => {
+    this.setState({ destination });
+    this._handleOnChange();
+  };
+
+  _handleCheckinChange = date => this._handleDateChange(date, 'checkin');
+
+  _handleCheckoutChange = date => this._handleDateChange(date, 'checkout');
+
+  _handleDateChange = async (date, type) => {
+    await this.setState({ [type]: date });
+    this._handleOnChange();
```

Comment:
I don't think the `await this.setState` works. I mean it works but not as you expect. You should use second parameter of the function in this case. Something like this:

```js
this.setState({ [type]: date }, () => this._handleOnChange());
```

Method `setState` [doesn't return `Promise`](https://github.com/facebook/react/blob/master/packages/react/src/ReactBaseClasses.js#L53). Correct me if I am wrong.